<launch>
  <arg name="mode" default="action" doc="mode type [debug, action]"/>

  <!-- Camera topic and weights, config and classes files -->
  <arg name="image_topic"                 default="/camera/sign_cam/sign_image_raw/compressed"/>
  <arg name="weights_name"	            default="yolov3-obj.weights"/>
  <arg name="config_name"	                default="yolov3-obj.cfg"/>
  <arg name="classes_name"	            default="obj.names"/>

  <!-- Published topics -->
  <arg name="publish_image"	            default="true"/>
  <arg name="detected_objects_topic"      default="detected_objects_in_image"/>
  <arg name="detections_image_topic"      default="detections_image_topic"/>

  <!-- Detection confidence -->
  <arg name="confidence"                  default="0.8"/>
  
    <node pkg="ecp_preproc" type="sign_detection.py" name="sign_detection" output="screen" respawn="true">
      <param if="$(eval mode == 'debug')" name="/is_debug_mode" value="True"/>
      <param if="$(eval mode == 'action')" name="/is_debug_mode" value="False"/>
      <param name="image_topic"               value="$(arg image_topic)"/>
      <param name="weights_name"              value="$(arg weights_name)"/>
      <param name="config_name"               value="$(arg config_name)"/>
      <param name="classes_name"              value="$(arg classes_name)"/>
      <param name="publish_image"             value="$(arg publish_image)"/>
      <param name="detected_objects_topic"    value="$(arg detected_objects_topic)"/>
      <param name="detections_image_topic"    value="$(arg detections_image_topic)"/>
      <param name="confidence"                value="$(arg confidence)"/>    
    </node>
    
</launch>
